compiler/ast_to_ir.py

```diff
@@ -159,7 +161,7 @@ def combine_pipe(ast_nodes):
     else:
         ## If any part of the pipe is not an IR, the compilation must fail.
         log("Node: {} is not pure".format(ast_nodes[0]))
-        raise Exception("Not pure node in pipe")
+        raise UnparallelizableError("Node: {} is not a pure node in pipe".format(ast_nodes[0]))
```

Comment:
Konstantinos (from old branch): I would include more information here (like the node), even though it is logged above.